### PR TITLE
Send notification info to app.ready, on macOS

### PR DESCRIPTION
**Pull request by CharlieHess**
This PR is an attempt to address https://github.com/electron/brightray/issues/168. Rather than clear out all notifications, we'll check for the `NSApplicationLaunchUserNotificationKey` and if found, send any notification arguments stashed in `userInfo` to the `app.ready` event. For Windows & Linux, this will always be an empty dictionary.

### CODE REVIEW

**Comment by paulcbetts** [comment]
#blessed

**Comment by felixrieseberg** [comment]
@zcbenz Let me know if there's anything I can do to get this over the :checkered_flag: line

File: atom/browser/atom_browser_main_parts.cc [hunk 1]
```diff
@@ -156,7 +156,8 @@ void AtomBrowserMainParts::PreMainMessageLoopRun() {
 #if !defined(OS_MACOSX)
   // The corresponding call in macOS is in AtomApplicationDelegate.
   Browser::Get()->WillFinishLaunching();
-  Browser::Get()->DidFinishLaunching();
+  base::DictionaryValue* empty_info = new base::DictionaryValue();
```
Comment (kevinsawicki) [inline atom/browser/atom_browser_main_parts.cc]: Will this be leaked since it is never freed and isn't wrapped in a smart pointer?

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**